src/mongocxx/bulk_write.cpp

```diff
@@ -84,6 +84,10 @@ void bulk_write::append(const model::write& operation) {
     }
 }
 
+bsoncxx::stdx::optional<void*> bulk_write::implementation() const {
+    return bsoncxx::stdx::optional<void*>(_impl.get());
```

Comment:
Do you actually need to name the return type here? Or can you just say return _impl.get()?